Library/Homebrew/mktemp.rb

```diff
@@ -62,9 +62,21 @@ def run
     begin
       Dir.chdir(tmpdir) { yield self }
     ensure
-      ignore_interrupts { rm_rf(tmpdir) } unless retain?
+      ignore_interrupts { chmod_rm_rf(tmpdir) } unless retain?
     end
   ensure
     ohai "Temporary files retained at:", @tmpdir.to_s if retain? && !@tmpdir.nil? && !@quiet
   end
+
+  private
+
+  def chmod_rm_rf(path)
+    if path.directory? && !path.symlink?
+      chmod("u+rw", path) # Need permissions in order to see the contents
```

Comment:
What if this fails?

Reply:
The current behaviour is it silently fails and stops the whole recursion. I imagine the only reason it could fail is if the path is not owned? I could add a `path.owned?` check in that case.

Reply:
A general `rescue` covering the whole function is probably a safe thing to do too, so if it does fail then it skips just that directory and nothing else.